src/elf/resolver.rs

```diff
@@ -80,7 +80,7 @@ impl ElfResolver {
         let loaded_address = if e_type == ET_EXEC {
```

Comment:
can't leave comment above, but there is no need to do a full loop, ELF specification says that PT_LOAD segments are ordered:

> PT_LOAD The array element specifies a loadable segment, described by p_filesz and
> p_memsz. The bytes from the file are mapped to the beginning of the memory
> segment. If the segment's memory size (p_memsz) is larger than the file size
> (p_filesz), the "extra'' bytes are defined to hold the value 0 and to follow the
> segment's initialized area. The file size may not be larger than the memory size.
> Loadable segment entries in the program header table appear in ascending order,
> sorted on the p_vaddr member.

Just mentioning another thing that could be improved as a follow up.

Reply:
Makes sense, thanks for pointing that out. Will keep that in mind when reworking this logic.